### PR TITLE
ENH: Minor change to parallel_coordinates (#15908)

### LINKED ISSUE #15908: Class label colours with parallel coordinates
#### Problem description
The parallel coordinates function seems to currently assume that the class labels don't have an inherent ordering. I'd like to provide a sort option or a class-to-colour mapping to capture this. My specific use case is situations where the label are quantized buckets and a smooth colour change between buckets make the final plot easy to visualize.

The current function zips the available classes and the colour_values without any sorting.

I've identified where in the current '_parallel_coordinates' function a change should be made and would like to work on this.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.4.2.final.0
python-bits: 64
OS: Linux
OS-release: 3.2.45-0.6.wd.865.49.315.metal1.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.19.2
nose: None
pip: 9.0.1
setuptools: 34.3.2
Cython: None
numpy: 1.12.1
scipy: 0.19.0
statsmodels: None
xarray: None
IPython: 5.3.0
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 2.0.0
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.999999999
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.5
boto: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by stangirala**
Add option to sort class lables, add to test

 - [x] closes #15908
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/plotting/test_misc.py [hunk 1]
```diff
@@ -241,6 +241,22 @@ def test_parallel_coordinates(self):
         with tm.assert_produces_warning(FutureWarning):
             parallel_coordinates(df, 'Name', colors=colors)
 
+        df = DataFrame({"feat": [i for i in range(30)],
```
Comment (jreback) [inline pandas/tests/plotting/test_misc.py]: make this a separate named test

Comment (jreback) [inline pandas/tests/plotting/test_misc.py]: add the issue number as a comment

File: pandas/tools/plotting.py [hunk 2]
```diff
@@ -718,6 +719,8 @@ def parallel_coordinates(frame, class_column, cols=None, ax=None, color=None,
         If true, vertical lines will be added at each xtick
     axvlines_kwds: keywords, optional
         Options to be passed to axvline method for vertical lines
+    sort_labels: bool, optional
+        Sort class_column labels, useful when assigning colours
```
Comment (jreback) [inline pandas/tools/plotting.py]: add a versionadded tag here, add the default (False) instead of optional

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
pls add a whatsnew entry, in Other Enhancements.

File: pandas/tests/plotting/test_misc.py (hunk 1 above)
Comment (stangirala) [inline pandas/tests/plotting/test_misc.py]: Done

**Comment by TomAugspurger** [comment]
@stangirala could you rebase and push one more time? Looks like our CI services were having trouble last time you pushed.

**Comment by stangirala** [comment]
@TomAugspurger @jreback Looks like the CI builds are passing.

**Comment by stangirala** [comment]
@TomAugspurger I'm actually not sure why the code coverage tool is reporting a decrease in coverage.

**Comment by TomAugspurger** [comment]
Yeah, I took a glance and couldn't figure it out either. I'm guessing it's just a fluke, but I'll take a closer look later and merge.

**Comment by stangirala** [comment]
Thanks!

**Comment by jreback** [comment]
if your master is not up to date code coverage doesn't make much sense

File: doc/source/whatsnew/v0.20.0.txt [hunk 3]
```diff
@@ -371,6 +371,8 @@ Other Enhancements
 - :func:`MultiIndex.remove_unused_levels` has been added to facilitate :ref:`removing unused levels <advanced.shown_levels>`. (:issue:`15694`)
 - ``pd.read_csv()`` will now raise a ``ParserError`` error whenever any parsing error occurs (:issue:`15913`, :issue:`15925`)
 - ``pd.read_csv()`` now supports the ``error_bad_lines`` and ``warn_bad_lines`` arguments for the Python parser (:issue:`15925`)
+- ``pd.read_csv()`` will now raise a ``csv.Error`` error whenever an end-of-file character is encountered in the middle of a data row (:issue:`15913`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.0.txt]: this line is duped

File: doc/source/whatsnew/v0.20.0.txt [hunk 4]
```diff
@@ -371,6 +371,8 @@ Other Enhancements
...
+- ``parallel_coordinates()`` now has a ``sort_labels`` keyword arg that sorts class labels and the colours assigned to them (:issue:`15908`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.0.txt]: ``has gained a ``

File: pandas/tests/plotting/test_misc.py [hunk 5]
```diff
@@ -241,6 +241,26 @@ def test_parallel_coordinates(self):
         with tm.assert_produces_warning(FutureWarning):
             parallel_coordinates(df, 'Name', colors=colors)
 
+    def test_parallel_coordinates_with_sorted_labels(self):
+        """ For #15908 """
+        from pandas.tools.plotting import parallel_coordinates
```
Comment (jreback) [inline pandas/tests/plotting/test_misc.py]: you may need a mpl skip here? @TomAugspurger ?

File: pandas/tools/plotting.py [hunk 6]
```diff
@@ -774,6 +780,9 @@ def parallel_coordinates(frame, class_column, cols=None, ax=None, color=None,
                                         colormap=colormap, color_type='random',
                                         color=color)
 
+    if sort_labels is True:
```
Comment (jreback) [inline pandas/tools/plotting.py]: ``if sort_labels`` is enough

**Comment by jreback** [comment]
lgtm. minor comment.

File: pandas/tests/plotting/test_misc.py (hunk 5 above)
Comment (TomAugspurger) [inline pandas/tests/plotting/test_misc.py]: Done on the class.

File: doc/source/whatsnew/v0.20.0.txt (hunk 3 above)
Comment (stangirala) [inline doc/source/whatsnew/v0.20.0.txt]: I don't see this duped on my copy, can you double check?

File: doc/source/whatsnew/v0.20.0.txt (hunk 4 above)
Comment (stangirala) [inline doc/source/whatsnew/v0.20.0.txt]: Fixed.

File: pandas/tools/plotting.py (hunk 6 above)
Comment (stangirala) [inline pandas/tools/plotting.py]: I mirrored an if expression from the lines above, but fixed.

**Comment by jreback** [comment]
thanks @stangirala

**Pull request closed without merging**